test/terraform_edns_cluster_test.go

```diff
@@ -0,0 +1,94 @@
+package test
+
+import (
+	"context"
+	"github.com/Azure/azure-sdk-for-go/services/dns/mgmt/2018-05-01/dns"
+	"github.com/google/uuid"
+	"github.com/gruntwork-io/terratest/modules/azure"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/otiai10/copy"
+	"github.com/stretchr/testify/assert"
+	"os"
+
+	"io/ioutil"
+	"path"
+	"testing"
+)
+
+func TestTerraformEDnsWithApexDomainCluster(t *testing.T) {
+
+	t.Parallel()
+
+	dirName, err := ioutil.TempDir("", uuid.New().String())
+	if err != nil {
+		t.Errorf("Failed to create temp dir %s", dirName)
+	}
+	defer os.Remove(dirName)
+
+	err = copy.Copy("../terraform_module", dirName)
+
+	if err != nil {
+		t.Errorf("Failed to copy terraform module to temp dir %s", dirName)
+	}
+
+	terraformOptions := &terraform.Options{
+		// Set the path to the Terraform code that will be tested.
+		TerraformDir: "../terraform_module",
```

Comment:
So, what I plan to do is have tests for all the example cases: 
https://github.com/jenkins-x/terraform-aws-eks-jx/tree/master/examples
Could be something you can do? Having readymade examples for end users would make it super easy for new users imo.

Reply:
Makes sense and I was already thinking along similar lines. My plan is to add to the test suite after this PR is merged and we have CI configured on PRs.